dataland-e2etests/src/test/kotlin/org/dataland/e2etests/auth/ApiKeyAuthenticationHelper.kt

```diff
@@ -0,0 +1,42 @@
+package org.dataland.e2etests.auth
+
+import org.dataland.datalandapikeymanager.openApiClient.api.ApiKeyControllerApi
+import org.dataland.datalandapikeymanager.openApiClient.model.ApiKeyAndMetaInfo
+import org.dataland.datalandapikeymanager.openApiClient.model.ApiKeyMetaInfo
+import org.dataland.datalandapikeymanager.openApiClient.model.RevokeApiKeyResponse
+import org.dataland.e2etests.BASE_PATH_TO_API_KEY_MANAGER
+
+class ApiKeyAuthenticationHelper {
+    private val jwtHelper = JwtAuthenticationHelper()
+    private val apiKeyManagerClient = ApiKeyControllerApi(BASE_PATH_TO_API_KEY_MANAGER)
+
+    fun obtainApikeyForTechnicalUser(technicalUser: TechnicalUser, daysValid: Int? = null): ApiKeyAndMetaInfo {
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(technicalUser)
+        val apiKeyAndMetaInfo = apiKeyManagerClient.generateApiKey(daysValid)
+        return apiKeyAndMetaInfo
+    }
+
+    fun authenticateApiCallsWithApiKeyForTechnicalUser
+    (technicalUser: TechnicalUser, daysValid: Int? = null): ApiKeyAndMetaInfo {
+        val metaInformation = obtainApikeyForTechnicalUser(technicalUser, daysValid)
+        GlobalAuth.setBearerToken(metaInformation.apiKey)
+        return metaInformation
+    }
+
+    fun revokeApiKeyForTechnicalUserAndResetAuthentication(technicalUser: TechnicalUser): RevokeApiKeyResponse {
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(technicalUser)
+        val revokeApiKeyResponse = apiKeyManagerClient.revokeApiKey()
+        GlobalAuth.setBearerToken(null)
+        return revokeApiKeyResponse
+    }
+
+    fun resetAuthenticationAndValidateApiKey(receivedApiKey: String): ApiKeyMetaInfo {
+        GlobalAuth.setBearerToken(null)
```

Comment:
Is this only done to check the reachability of this endpoint without any authorities? Technically the bearer token could remain set. I just found it quite weird to read the name of this function so often in the DataRetrievalViaApiKeyTest.kt